src/protocol.py

```diff
@@ -262,7 +262,8 @@ def assembleVersionMessage(remoteHost, remotePort, participatingStreams, server=
         payload += encodeHost('127.0.0.1')
         payload += pack('>H', 8444)
     else:
-        payload += encodeHost(remoteHost)
+        # 127.0.0.1 for onion v3 services
+        payload += encodeHost(remoteHost if len(remoteHost) <= 22 else '127.0.0.1')
```

Comment:
remoteHost is helpful in the logs so that you can see your external IP address. Maybe it's not currently used other than in the logs but it could in the future.

Reply:
So what to do? Place there first 16 bytes of onion v3 pubkey?

Reply:
I think that would be better than 127.0.0.1